telemetry/definitions/commonDefinitions.json

```diff
@@ -51,7 +51,7 @@
             "description": "The Lambda Package type of the function"
         },
         {
-            "name": "lambdaArchitecture",
+            "name": "architecture",
```

Comment:
Renaming this because it is identical with the `architecture` type, which is used in several other metrics. Should be easy to update this in all toolkits @rli @awschristou

Reply:
`lambdaArchitecture` was defined in-product Sept 2021 (the same time as the VS Code Toolkit change). It was migrated over in Feb 2022. There was an internal discussion about this before adding lambdaArchitecture into the definitions.

Reply:
`architecture` has been used in 

- `apigateway_invokeLocal`
- `lambda_invokeLocal`
- `sam_init`
- `sam_attachDebugger` (which will likely be removed)

, which are now being lifted into common (previously was local to vscode).

We will have outdated metrics whichever choice we make. Or I could add a redundant field (`lambdaArchitecture`) to those metrics.

I'll do that since there is concern about the current PR.

Reply:
Personally, I'd let both types exist but not add redundant fields. This is the only true non-breaking change from a consumer's perspective. Otherwise you will have fragmented the data-set.

The fundamental problem here is that our telemetry design associates data types 1:1 with field names. This goes against most modern IDLs (think Smithy, OpenAPI, WADL) as well as pretty much every strongly-typed programming language that supports interfaces.

Names of fields describe additional context applicable to the associated data type. But right now we cannot express that. The information is bound to the data type itself.

Reply:
> The datastore doesn't care about types. So we should use consistent names.

Our datastore actually _does_ care about types, we just automatically make everything a string :)
But yeah consistent naming is important. It's just that our current design gets in the way of sharing types with the same semantics but not the same context.

> An alternative is to have both names and drop the deprecated one a year from now. That is what this PR aims to do.

Yup I see what your intent was now. For some reason I thought we were going to have Toolkits emit one or the other, not both. If both are emitted, then that works.